c_glib/test/test-extension-data-type.rb

```diff
@@ -102,4 +102,11 @@ def test_wrap_chunked_array
                    extension_chunked_array.chunks.collect(&:class),
                  ])
   end
+
+  # GH-44478 test https://github.com/apache/arrow/issues/44478
+  def test_abstract_class_init
+    assert_raise(TypeError) do
+      Arrow::ExtensionDataType.new.name
```

Comment:
Do we need this `.name`?

```suggestion
      Arrow::ExtensionDataType.new
```

Reply:
* `Arrow::ExtensionDataType.new`: Does not segmentation fault.
* `Arrow::ExtensionDataType.new.name`:  segmentation fault.

That's why I added `.name`.

Reply:
OK. But the new implementation raises by `Arrow::ExtensionDataType.new` and `.name` is never called, right?
If so, we don't need `.name` here.

Reply:
Thanks. You are right the new implementation raise the exception by `Arrow::ExtensionDataType.new`. I removed `.name`.